### DISCUSSION TITLE
ContextMenu Connect & Disconnect methods

### DISCUSSION THREAD

**Discussion by BayouBushBaby** [FirstGearGames/FishNet#457] [Feature Request]
Add context menu functions to TransportManager to allow quick connecting/disconnecting in editor for debugging.
`#if UNITY_EDITOR
    
    //ContextMenu Methods
    [ContextMenu("StartConnection/Server")]
    private void ContextMenu_ConnectAsServer()
        => InstanceFinder.TransportManager.Transport.StartConnection(true);
    
    [ContextMenu("StartConnection/Client")]
    private void ContextMenu_ConnectAsClient()
        => InstanceFinder.TransportManager.Transport.StartConnection(false);
    
    [ContextMenu("StopConnection/Server")]
    private void ContextMenu_DisconnectAsServer()
        => InstanceFinder.TransportManager.Transport.StopConnection(true);
    
    [ContextMenu("StopConnection/Client")]
    private void ContextMenu_DisconnectAsClient()
        => InstanceFinder.TransportManager.Transport.StopConnection(false);
    
    [ContextMenu("StartConnection/Server", isValidateFunction:true)]
    private bool ContextValidate_ConnectAsServer()
    {
        if (!EditorApplication.isPlaying)
            return false;
        
        var networkManager = InstanceFinder.NetworkManager;
        return networkManager &&  networkManager.IsOffline && !networkManager.IsServer;
    }
    
    //ContextMenu Validation Methods
    [ContextMenu("StartConnection/Client", isValidateFunction:true)]
    private bool ContextValidate_ConnectAsClient()
    {
        if (!EditorApplication.isPlaying)
            return false;
        
        var networkManager = InstanceFinder.NetworkManager;
        return networkManager && !networkManager.IsOffline && !networkManager.IsClient;
    }
    
    [ContextMenu("StopConnection/Server", isValidateFunction:true)]
    private bool ContextValidate_DisconnectAsServer()
    {
        if (!EditorApplication.isPlaying)
            return false;
        
        var networkManager = InstanceFinder.NetworkManager;
        return networkManager && !networkManager.IsOffline && networkManager.IsServer;
    }
    
    [ContextMenu("StopConnection/Client", isValidateFunction:true)]
    private bool ContextValidate_DisconnectAsCliente()
    {
        if (!EditorApplication.isPlaying)
            return false;
        
        var networkManager = InstanceFinder.NetworkManager;
        return networkManager && !networkManager.IsOffline && networkManager.IsClient;
    }
    
    #endif`

**Comment by BayouBushBaby** [comment]
For the love of god I can't get the "#if UNITY_EDITOR" inside the blog but you get the point.

**Comment by FirstGearGames** [comment]
Wouldn't we want to use the networkmanager reference rather than InstanceFinder?

**Comment by FirstGearGames** [comment]
I'd like to keep as much as possible out of the inspector. I thought about this for awhile but decided if anything a custom component should be made, rather than implementing this directly inside the TransportManager.
